### ISSUE TITLE
Publisher_acl authentication error

### ISSUE THREAD

**Issue by frogunder** [saltstack/salt#45023]
Testing `publisher_acl system` on the Oxygen branch fails.

I installed with the bootstrap script using this commit `3ed7a29c56e1ce9959113138c27fd601093a8e56`

I get this error 
```
Failed to authenticate! This is most likely because this user is not permitted to execute commands, but there is a small possibility that a disk error occurred (check disk/inode usage).
```

I see this in the master log:
```
2017-12-14 21:34:43,817 [salt.auth        :332 ][WARNING ][15612] Authentication failure of type "user" occurred.
2017-12-14 21:34:43,818 [salt.master      :1909][WARNING ][15612] Authentication failure of type "user" occurred.
```

I also see this in the master log:
```
2017-12-14 21:32:31,794 [salt.master      :617 ][ERROR   ][15589] Unable to load SSDP: asynchronous IO is not available.
2017-12-14 21:32:31,796 [salt.master      :619 ][ERROR   ][15589] You are using Python 2, please install "trollius" module to enable SSDP discovery.
```
Installing `trollius` did not help.

### Setup
Put this in the master config file:
```
publisher_acl:
  testuser:
    - .*
```

Add user and su into that user.

Modify the permission.
https://docs.saltstack.com/en/develop/ref/publisheracl.html

### Steps to Reproduce Issue
Try to run a command as the new user will fail with the `Failed to authenticate` message.

### Versions Report
```
salt --versions
Salt Version:
           Salt: 2018.2.0-22-g3ed7a29
 
Dependency Versions:
           cffi: 1.10.0
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.9.6
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.14
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.14 (default, Nov  3 2017, 10:55:25)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.2
            ZMQ: 4.1.6
 
System Versions:
           dist: fedora 27 Twenty Seven
         locale: UTF-8
        machine: x86_64
        release: 4.13.9-300.fc27.x86_64
         system: Linux
        version: Fedora 27 Twenty Seven
```

**Comment by frogunder** [comment]
I tried this again and I still get the error. 

I did I git bisect and found this:
```
20bf4eed1d34cf8edfa41a73a9769ffa7a977449 is the first bad commit
commit 20bf4eed1d34cf8edfa41a73a9769ffa7a977449
Author: kstreee <kstreee@gmail.com>
Date:   Fri Oct 20 11:06:31 2017 +0900

    Separates key_dir from cache_dir, The key files (i.e. '.root_key', '.sudo_...') must not be shared with other masters.

:040000 040000 36c7b2997235e09f243be28a0c714c8c6b04efe5 9e141e6c68ddd2d4b8ce846a7a95303fe76e381c M	salt
:040000 040000 e14322a871691753ab8d5f95fbd8ccb9bf473c5b d21b3b1d7bd736b66d6259f70af5537c1aea0c76 M	tests
```

It is working correctly on the commit before the one mentioned above.
Also this is working fine on 2017.7.2.

**Comment by frogunder** [comment]
Tried this again on the head of Oxygen.rc1 (fcf8a36a4aa61ea017846a278b7fbfcceefa1d14) and I get the authentication error. I do not see the other error only the authentication error.
```
[testuser@localhost root]$ salt \* test.ping
Authentication error occurred.
[testuser@localhost root]$ 
```
In the master log
```
2018-02-06 17:41:49,120 [salt.auth        :333 ][WARNING ][17156] Authentication failure of type "user" occurred.
2018-02-06 17:41:49,121 [salt.master      :2045][WARNING ][17156] Authentication failure of type "user" occurred.
```
```
Salt Version:
           Salt: 2018.2.0-1788-gfcf8a36
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.1
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.5 (default, Aug  4 2017, 00:39:18)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: centos 7.4.1708 Core
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-693.el7.x86_64
         system: Linux
        version: CentOS Linux 7.4.1708 Core
```

**Comment by rallytime** [comment]
@frogunder This looks similar to the fix in https://github.com/saltstack/salt/pull/45811, but that fix isn't _quite_ right for this problem. I have assigned this to @gtmanfred to look into. Thanks!

**Comment by rallytime** [comment]
@frogunder This should be fixed with #45931. Can you confirm?

**Comment by frogunder** [comment]
@rallytime It is working fine for me now, with that fix.
